extensions/amp-instagram/1.0/test/test-instagram.js

```diff
@@ -61,39 +63,68 @@ describes.sandboxed('Instagram preact component v1.0', {}, (env) => {
     expect(wrapper.find('div')).to.have.lengthOf(2);
   });
 
-  it('Resize prop is called', async () => {
-    const wrapper = mount(
+  it('Resize prop is called', () => {
+    mount(
       <Instagram
         shortcode="B8QaZW4AQY_"
-        style={{'width': 500, 'height': 705}}
+        style={{'width': 500, 'height': 600}}
         requestResize={env.sandbox.spy()}
       />,
       {attachTo: document.body}
     );
 
-    await new Promise((resolve) =>
-      wrapper.find('iframe').instance().addEventListener('load', resolve)
+    const requestResizeSpy = env.sandbox.spy();
+    const heightSpy = env.sandbox.spy();
+    const opacitySpy = env.sandbox.spy();
+
+    handleMessage(
+      createMockEvent(),
+      useRef({'contentWindow': 'source'}),
+      requestResizeSpy,
+      heightSpy,
+      opacitySpy
     );
-    expect(wrapper.prop('requestResize')).to.have.been.calledOnce;
+
+    expect(requestResizeSpy).to.have.been.calledOnce;
+    expect(heightSpy).to.have.not.been.called;
+    expect(opacitySpy).to.have.been.calledOnce;
   });
 
-  it("Container's height is changed", async () => {
-    const initialHeight = 300;
-    document.body.addEventListener('message', console.log);
-    const wrapper = mount(
+  it('Height is changed', () => {
+    mount(
       <Instagram
         shortcode="B8QaZW4AQY_"
-        style={{'width': 500, 'height': initialHeight}}
+        style={{'width': 500, 'height': 600}}
+        requestResize={env.sandbox.spy()}
       />,
       {attachTo: document.body}
     );
 
-    await new Promise((resolve) =>
-      wrapper.find('iframe').instance().addEventListener('load', resolve)
+    const heightSpy = env.sandbox.spy();
+    const opacitySpy = env.sandbox.spy();
+
+    handleMessage(
+      createMockEvent(),
+      useRef({'contentWindow': 'source'}),
+      null,
+      heightSpy,
+      opacitySpy
     );
 
-    expect(
-      wrapper.find('iframe').instance().parentElement.parentElement
-    ).to.not.equal(initialHeight);
+    expect(heightSpy).to.have.been.calledOnce;
+    expect(opacitySpy).to.have.been.calledOnce;
   });
 });
+
+function createMockEvent() {
+  const mockEvent = new CustomEvent('message');
```

Comment:
Can these init properties be passed as an options object to the constructor here? ie.
```javascript
return new CustomEvent('message', { ... });
```

Reply:
Unfortunately not, the CustomEventInit object only takes a field named "detail". https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent